app/views/qing_groups/_group_inner.html.erb

```diff
@@ -1,4 +1,4 @@
-<div class="inner">
+<div class="inner" href="<%= read_only ? qing_group_path(qing_group) : edit_qing_group_path(qing_group) %>">
```

Comment:
We have a QingGroupDecorator you could put this big long thing in...

Reply:
But please try to call .decorate only once. Could decorate when passing into this partial.